lightning-invoice/src/utils.rs

```diff
@@ -792,6 +812,55 @@ mod test {
 		assert_eq!(invoice.payment_hash(), &sha256::Hash::from_slice(&payment_hash.0[..]).unwrap());
 	}
 
+	#[test]
+	fn test_hints_has_only_public_confd_channels() {
```

Comment:
I think it would make sense to also add test coverage that ensures that the announced channel gets removed from the route hints once it has 7 confirmations, no?